src/style/properties.js

```diff
@@ -349,6 +349,7 @@ const styfn = {};
     { name: 'taxi-turn', type: t.bidirectionalSizeMaybePercent, triggersBounds: diff.any },
     { name: 'taxi-turn-min-distance', type: t.size, triggersBounds: diff.any },
     { name: 'taxi-direction', type: t.axisDirection, triggersBounds: diff.any },
+    { name: 'taxi-curve', type: t.bool, triggersBounds: diff.any },
```

Comment:
The first thing other people who would use this feature would ask is:  How do I set the radius of the curve, [analogous to `border-radius` in CSS](https://developer.mozilla.org/en-US/docs/Web/CSS/border-radius)?

With a small radius, the effect of the PR is nearly indistinguishable from the existing, [curved line-join](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/lineJoin).  Some people will certainly want a subtle curve, but many people will want a stronger one.

With a configurable radius, you could get away with using a single property (e.g. `taxi-turn-radius`, default 0).  In that case, 0 would be equivalent to your proposed `taxi-curve: no`.